### PR TITLE
minor improvement

### PR DESCRIPTION
**Pull request by leowucn**
The `s.mcpOptions` assignment snippet would be unnecessarily run multiple times in case the length of `s.mesh.ConfigSources` is greater than 1.

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -747,18 +751,15 @@ func (s *Server) mcpController(args *PilotArgs,
 	clients *[]*sink.Client,
 	configStores *[]model.ConfigStoreCache) {
 	clientNodeID := ""
-	var collections []sink.CollectionOptions
+	collections := make([]sink.CollectionOptions, len(schemas.Istio)-1)
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go]: this is not right, this will result in empty indexes at the start

Comment (leowucn) [reply pilot/pkg/bootstrap/server.go]: Yes, you are right, and I made a change

**Pull request merged**